src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -267,6 +267,9 @@ export class HttpConnection implements IConnection {
                         // the returned access token
                         const accessToken = negotiateResponse.accessToken;
                         this._accessTokenFactory = () => accessToken;
+                        // set the factory to undefined so the httpClient won't retry with the same token, since we know it won't change until a connection restart
+                        this._httpClient._accessToken = accessToken;
+                        this._httpClient._setAccessTokenFactory(undefined);
```

Comment:
Remind me why this isn't just:

```suggestion
        this._httpClient._accessTokenFactory = undefined;
```

similar to what we're doing for `_httpClient._accessToken`?

Reply:
I think David wanted it to be a function so we're not modifying the field. Should probably change `_accessToken` to also be modified via a function.

Reply:
Just so it's not "readable" @davidfowl? This isn't really encapsulated anyway. This is all internal, but it's JS/TS so anyone could monkey patch. Adding setters for these just obscures what's going on. It makes it look like it might be doing something more complicated.

Reply:
Ok, looking at the history, it was actually me that made it a function 😆 and David just suggested a better name for the func.